lib/vector-core/src/event/metadata.rs

```diff
@@ -70,6 +73,16 @@ impl EventMetadata {
         &mut self.secrets
     }
 
+    /// Returns a mutable reference to the metadata source.
+    pub fn source(&self) -> &Option<Arc<OutputId>> {
```

Comment:
It may be useful to add convenience functions to `LogEvent` etc with the same names.